src/util/type-utils.ts

```diff
@@ -153,19 +153,19 @@ export type Equals<T, U> =
     ? true
     : false
 
-export type NarrowPartial<O, T> = DrainOuterGeneric<
```

Comment:
The `DrainOuterGeneric` never should've been wrapped around the conditional type; it only makes sense on the object type in the true branch. But `NarrowPartial` didn't used to be a conditional type, and then I (in a long ago PR) made `NarrowPartial` conditional but didn't move the `DrainOuterGeneric`.